02 - Web UI Template/CognitiveSearch.UI/Search/DocumentSearchClient.cs

```diff
@@ -337,67 +337,71 @@ public DocumentResult GetDocuments(string q, SearchFacet[] searchFacets, int cur
-            tokens[0] = container.GetSharedAccessSignature(adHocPolicy, null);
-
-            // Get token for second indexer data source
-            containerAddresses[1] = _configuration.GetSection("StorageContainerAddress2")?.Value.ToLower();
-            if (!String.Equals(containerAddresses[1], defaultContainerUriValue))
+            StorageSharedKeyCredential storageSharedKeyCredential = new StorageSharedKeyCredential(accountName, accountKey);
+            s_containerAddresses[0] = _configuration.GetSection("StorageContainerAddress")?.Value.ToLower();
+            s_containerAddresses[1] = _configuration.GetSection("StorageContainerAddress2")?.Value.ToLower();
+            s_containerAddresses[2] = _configuration.GetSection("StorageContainerAddress3")?.Value.ToLower();
+            int s_containerAddressesLength = s_containerAddresses.Length;
+            if (!String.Equals(s_containerAddresses[1], defaultContainerUriValue))
             {
-                CloudBlobContainer container2 = new CloudBlobContainer(new Uri(containerAddresses[1]), new StorageCredentials(accountName, accountKey));
-                tokens[1] = container2.GetSharedAccessSignature(adHocPolicy, null);
+                s_containerAddressesLength--;
             }
-
-            // Get token for third indexer data source
-            containerAddresses[2] = _configuration.GetSection("StorageContainerAddress3")?.Value.ToLower();
-            if (!String.Equals(containerAddresses[2], defaultContainerUriValue))
+            if (!String.Equals(s_containerAddresses[2], defaultContainerUriValue))
             {
-                CloudBlobContainer container3 = new CloudBlobContainer(new Uri(containerAddresses[2]), new StorageCredentials(accountName, accountKey));
-                tokens[2] = container3.GetSharedAccessSignature(adHocPolicy, null);
+                s_containerAddressesLength--;
             }
+            for (int i = 0; i < s_containerAddressesLength; i++) {
+                BlobContainerClient container = new BlobContainerClient(new Uri(s_containerAddresses[i]), new StorageSharedKeyCredential(accountName, accountKey));
+                var policy = new BlobSasBuilder
+                {
+                    Protocol = SasProtocol.HttpsAndHttp,
+                    BlobContainerName = container.Name,
+                    Resource = "c",
+                    StartsOn = DateTimeOffset.UtcNow,
+                    ExpiresOn = DateTimeOffset.UtcNow.AddHours(24),
+                    IPRange = new SasIPRange(IPAddress.None, IPAddress.None)
+                };
+                policy.SetPermissions(BlobSasPermissions.Read);
+                var sas = policy.ToSasQueryParameters(storageSharedKeyCredential);
+                BlobUriBuilder  sasUri = new BlobUriBuilder(container.Uri);
+                sasUri.Sas = sas;
 
-            return tokens;
+                s_tokens[i] = sasUri.ToUri().ToString();
```

Comment:
This line doesn't work based on the rest of the logic in the app - it causes the sas token + url to be appended to the url. 

I believe this should be changed to 
```csharp
s_tokens[i] = "?" + sasUri.Sas.ToString();
```
Please confirm that sas tokens work in the UI